### PR TITLE
Sa 176 - Added DeleteVolume function and renamed the existing DeleteVolume to DeestroyVolumee 

### PR DESCRIPTION
**Pull request by vidhutsingh**
Fix : 
Added DeleteVolume which doesnt make the volume offline and tries to delete the volume . It is expected from user to ensure volume is offline . wherein DestroyVolume makes the volume offline prior to delete .

### CODE REVIEW

File: pkg/service/volume_service.go [hunk 1]
```diff
@@ -145,10 +145,10 @@ func (svc *VolumeService) OfflineVolume(id string, force bool) (*nimbleos.Volume
 
 }
 
-// DeleteVolume - deletes the volume
-func (svc *VolumeService) DeleteVolume(id string) error {
+// DestroyVolume - deletes the volume
```
Comment (rkumpf) [inline pkg/service/volume_service.go]: The description does not distinguish it from DeleteVolume.

File: pkg/service/snapshot_service.go [hunk 2]
```diff
@@ -72,13 +72,13 @@ func (svc *SnapshotService) GetSnapshotById(id string) (*nimbleos.Snapshot, erro
 	return snapshotResp, nil
 }
 
-// GetSnapshotByName - method returns a pointer "Snapshot"
-func (svc *SnapshotService) GetSnapshotByName(name string) (*nimbleos.Snapshot, error) {
+// GetSnapshotByName - method returns a list of pointers to "Snapshot"
+func (svc *SnapshotService) GetSnapshotByName(volName string) ([]*nimbleos.Snapshot, error) {
```
Comment (rkumpf) [inline pkg/service/snapshot_service.go:76]: The function name is confusing.  It should be GetSnapshotsByVolumeName.

**Comment by rkumpf** [comment]
Please add ckallianpur to review the fakeservice code.

File: pkg/service/snapshot_service.go (hunk 2 above)
Comment (vidhutsingh) [reply pkg/service/snapshot_service.go:76]: This cannot be handled with this review. please don't mix.

File: pkg/service/volume_service.go (hunk 1 above)
Comment (vidhutsingh) [reply pkg/service/volume_service.go]: sure .

**Comment by vidhutsingh** [comment]
> Please add ckallianpur to review the fakeservice code.

This is not fakeservice changes

File: pkg/service/snapshot_service.go (hunk 2 above)
Comment (vidhutsingh) [reply pkg/service/snapshot_service.go:76]: There is going to be next PR generated from build . you should expect the fix there .

Comment (rkumpf) [reply pkg/service/snapshot_service.go:76]: I don't understand the "don't mix" comment.  If generated things are out of bounds, what is left to review?  Anyway, if you are saying the change is coming in the next PR, that is fine with me.

**Comment by rkumpf** [comment]
> > Please add ckallianpur to review the fakeservice code.
> 
> This is not fakeservice changes

Was the fakeservice code included in this PR by mistake?

File: pkg/service/snapshot_service.go (hunk 2 above)
Comment (vidhutsingh) [reply pkg/service/snapshot_service.go:76]: The fix is only for specific issues. This PR doesn't contain any changes related to the snapshot.

**Comment by vidhutsingh** [comment]
> > > Please add ckallianpur to review the fakeservice code.
> > 
> > 
> > This is not fakeservice changes
> 
> Was the fakeservice code included in this PR by mistake?

seems like fakeservice is included by mistake. will removed it from this PR

File: pkg/service/snapshot_service.go (hunk 2 above)
Comment (rkumpf) [reply pkg/service/snapshot_service.go:76]: Maybe the snapshot changes were also included by mistake.

**Review by rkumpf — APPROVED** [review]

**Pull request merged**